### PR TITLE
handle the "we couldn't find that photo issue"

### LINKED ISSUE #18: Photo: We couldn't find that photo
First of all, let me know that this tool is awesome, thanks for it.
I'm putting this issue here to discuss it, I'll propose a solution via a pull request if you find this is appropriate.

My use case:
I'm on ubuntu and I use a cron that call a shell script (to setup all the info needed to access the current wallpaper ubuntu service) that runs your script.
I change the picture every 5 minutes (night & day 24/7 ...)
So I have a lot of chance to come across an issue more than someone that run it manually daily.

This time I got the picture: https://images.unsplash.com/photo-1446704477871-62a4972035cd?fit=crop&fm=jpg&h=800&q=50&w=1200 (or a similar one, this one is from a random google image search)

So what would have been expected (from my point of view): to have an error that I can handle further on in my script

### PR DESCRIPTION
**Pull request by scips**
This is my proposal to handle the:  "we couldn't find that photo issue" and fixes #18 
There is no clear documentation and a better approach could be to use the API but according to some search on google it seems that the 404 image remains the same.

However I was not able to run: npm run lint without issue. I'm maybe missing a configuration file as everything seems wrong. So I didn't change indentation and left it as it is.

Regards,

Sébastien

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -99,6 +99,12 @@ exports.download = (opts, url, reporter) => {
         });
         return;
       }
+      if (res.request.uri.pathname.indexOf('photo-1446704477871-62a4972035cd') >= 0) {
+        reject({
+          message: 'The response is the image: we couldn\'t find that photo.'
+        });
+        return;
+      }
```
Comment (cuth) [inline lib/index.js]: This seems as good as we can probably make it. If they change the 404 image, the worst that can happen is this code doesn't run.

File: test/lib.js [hunk 2]
```diff
@@ -232,3 +232,25 @@ test('download', async t => {
...
+test('download error', async t => {
+  const options = {
+    dir: __dirname
+  };
+  const url = 'https://source.unsplash.com/WLUHO9A_xiki/1600x900'
+  const stateStack = [];
+  return download(options, url, state => stateStack.push(state)).catch((err)=>{
+    t.is(err.message, 'The response is the image: we couldn\'t find that photo.');
+  });
+});
```
Comment (cuth) [inline test/lib.js]: Can we add a newline character at the end of the file?

File: test/lib.js [hunk 3]
```diff
@@ -232,3 +232,25 @@ test('download', async t => {
...
   return pify(fs.unlink)(file);
 });
+
+test('download not an image', async t => {
+  const options = {
+    dir: __dirname
+  };
+  const url = 'https://source.unsplash.com/'
+  const stateStack = [];
+  return download(options, url, state => stateStack.push(state)).catch((err)=>{
```
Comment (cuth) [inline test/lib.js]: I know you pulled this from my example above (thank you for keeping a similar code style), but I have no idea what `stateStack` does. Do you? I can't remember what it was for.

Comment (scips) [reply test/lib.js]: stateStack contains the progress in % but I think it's useless in the test. I'll remove it.

**Comment by cuth** [comment]
Thanks, @scips!

**Comment by cuth** [comment]
This is released in 6.2.0

**Pull request merged**